object/object.go

```diff
@@ -623,88 +631,43 @@ func (o *Object) PreviousID() (oid.ID, bool) {
 //
 // See also [Object.SetPreviousID].
 func (o Object) GetPreviousID() oid.ID {
-	var id oid.ID
-	if m := (*object.Object)(&o).GetHeader().GetSplit().GetPrevious(); m != nil {
-		if err := id.ReadFromV2(*m); err != nil {
-			panic(fmt.Errorf("unexpected ID decoding failure: %w", err))
-		}
-	}
-	return id
+	return o.header.split.prev
 }
 
 // ResetPreviousID resets identifier of the previous sibling object.
 //
 // See also [Object.SetPreviousID], [Object.GetPreviousID].
 func (o *Object) ResetPreviousID() {
-	o.setSplitFields(func(split *object.SplitHeader) {
-		split.SetPrevious(nil)
-	})
+	o.header.split.prev = oid.ID{}
 }
 
 // SetPreviousID sets identifier of the previous sibling object.
 //
 // See also [Object.GetPreviousID].
 func (o *Object) SetPreviousID(v oid.ID) {
-	var v2 refs.ObjectID
-	v.WriteToV2(&v2)
-
-	o.setSplitFields(func(split *object.SplitHeader) {
-		split.SetPrevious(&v2)
-	})
+	o.header.split.prev = v
 }
 
 // Children return list of the identifiers of the child objects.
 //
 // See also [Object.SetChildren].
 func (o Object) Children() []oid.ID {
-	v2 := (*object.Object)(&o)
-	ids := v2.GetHeader().GetSplit().GetChildren()
-
-	var (
-		id  oid.ID
-		res = make([]oid.ID, len(ids))
-	)
-
-	for i := range ids {
-		if err := id.ReadFromV2(ids[i]); err != nil {
-			panic(fmt.Errorf("unexpected child#%d decoding failure: %w", i, err))
-		}
-		res[i] = id
-	}
-
-	return res
+	return o.header.split.children
 }
 
 // SetChildren sets list of the identifiers of the child objects.
 //
 // See also [Object.Children].
 func (o *Object) SetChildren(v ...oid.ID) {
-	var (
-		v2  refs.ObjectID
-		ids = make([]refs.ObjectID, len(v))
-	)
-
-	for i := range v {
-		v[i].WriteToV2(&v2)
-		ids[i] = v2
-	}
-
-	o.setSplitFields(func(split *object.SplitHeader) {
-		split.SetChildren(ids)
-	})
+	o.header.split.children = v
```

Comment:
copying was not stated in docs, so nobody should rely on this

Reply:
relying on undocumented behavior is a user mistake. And it cannot be fixed by some new docs

Reply:
People do mistakes whether you like it or not. What's not obvious here is whether this setter makes a copy or not, it can be implemented in any way and you're changing it. People wouldn't be asking questions if they'd be doing
```
o.Childer = smth
```
but they do
```
o.SetChildren(smth...)
```
Take https://pkg.go.dev/bytes#Cut and https://pkg.go.dev/bytes#ToLower, both can be implemented with copying and without copying, but you know what to expect from the documentation.

Reply:
> People do mistakes whether you like it or not.

im ok with this

> Take https://pkg.go.dev/bytes#Cut and https://pkg.go.dev/bytes#ToLower

these examples accept and returns slices, whether the same or not is a completely natural question. I dont think it's worth to write this about any func with slice element. Copying is an unnecessary expense for some cases, they are worth writing about. Here is no copy, and so no docs. Even if i write, somebody can copy again anyway

Reply:
https://pkg.go.dev/math/big@go1.23.4#Rat.Set
https://pkg.go.dev/math/big@go1.23.4#Int.SetBits
https://pkg.go.dev/net/url@go1.23.4#URL.ResolveReference

A big part of the problem is setters/getters pattern itself.

Reply:
u may try to intro these docs on ur own, this is unrelated to PR theme

getters and setters is the only thing making this change backward compatible. And they make code representative, and inlining make them free

Reply:
You're changing the behavior in this PR, that's why this thread was started in the first place.

Reply:
yeah, cuz im free to change it according to docs, nothing what declared is violated

Reply:
No wonder people have problems with NeoFS SDK.

Reply:
sad to know this, motivating at the same time. We'll try our best to improve usability and help users within #224. I think the key improvement would be a set of practically useful examples. Docs are also important ofc